src/main/java/edu/aau/groupc/canteenbackend/user/User.java

```diff
@@ -0,0 +1,63 @@
+package edu.aau.groupc.canteenbackend.user;
+
+import edu.aau.groupc.canteenbackend.entities.DBEntity;
+import lombok.Data;
+
+import javax.persistence.*;
+
+@Data
+@Entity
+@Table(name = "users", uniqueConstraints = {
+        @UniqueConstraint(columnNames = {"username"}),
+})
+public class User implements DBEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    private String username;
+    private String password;
```

Comment:
Security consideration: Only ever save passwords hashed. 
(Of course this is not really relevant for our Uni project, but it should still be quite straightforward to implement.)

Reply:
Sure, this is only a basic implementation so that others can see what can they expect (even before it gets merged). Passwords and validations are not so important for this course but, of course, we will do basic password hashing and form validation.

REST-API in Spring seems to be really easy, especially after having an example you created for `Dish`.
The most confusing part was with `JpaRepositories`. Didn't know they 'detect' queries by interface methods' names.

When it comes to design, every other part of the project will probably have to use my `AuthService` to check whether the user is authorized to do an action **on every single request**. Do you think this design is ok, or have any better ideas?

Reply:
In my opinion the authentication mechanism is well designed.

Afaik, we could implement a custom Annotation for endpoint methods which authenticates the user based on the token.

Something like this:
`@RequireRole(ADMIN)
Response someAdminAction(...)`

I'll have a look into that.

Reply:
That would be cool. Is that feature (creating custom annotations) specific to Spring only or a general Java feature?
Reply here if you find anything specific.

Reply:
I actually just got it working a minute ago. I've pushed my intermediate results (no tests yet, but seems to work) to [feature/endpoint-security](https://github.com/AAU-ASE-GroupC-WS2021/canteenMgmtBackend/tree/feature/endpoint-security) so you can have a look if you're interested. I had to change a few small details about your implementation - mostly just so I could find the a user / auth by just the token (so without username). 

Maybe after I write some tests we can just merge that branch into your `feature/user-authentication` since you might have to make some minor adjustments because of those changes I had to make.

As for custom annotations, that is a standard Java feature. You can create just "plain" Meta annotations or link them to a processor to handle actual logic.